core/src/main/java/cucumber/runtime/formatter/PrettyFormatter.java

```diff
@@ -208,22 +208,27 @@ private void printStep(TestStep testStep, Result result) {
     }
 
     String formatStepText(String keyword, String stepText, Format textFormat, Format argFormat, List<Argument> arguments) {
-        int textStart = 0;
+        int beginIndex = 0;
         StringBuilder result = new StringBuilder(textFormat.text(keyword));
         for (Argument argument : arguments) {
             // can be null if the argument is missing.
             if (argument.getOffset() != null) {
-                String text = stepText.substring(textStart, argument.getOffset());
+                int argumentOffset = argument.getOffset();
+                if (argumentOffset < beginIndex ) { // a nested argument starts before the enclosing argument ends; ignore it when formatting
```

Comment:
It is okay to place line comments above the block or line they reference. Especially if the comment is long.